charts/argo-workflows/templates/controller/workflow-sa.yaml

```diff
@@ -1,13 +1,31 @@
 {{- if .Values.workflow.serviceAccount.create -}}
+
+  {{- /*
```

Comment:
I am not certain this condition check is worth it here. But if you insist on that consider that both `.Values.workflow.namespace` and `.Values.controller.workflowNamespaces` are only effective when `.Values.singleNamespace` is `false`. Otherwise, everything happens in the controller's namespace.

Same aplies for the type enforcement below. Pluse the users will be using the default value of `.Values.controller.workflowNamespaces` in `values.yaml` as a reference and it is already a list. And of course, the `concat` function in the files above will fail if it is not.

Reply:
Agreed